src/convertTypesGTM.ts

```diff
@@ -0,0 +1,70 @@
+import { EventPayload } from './EventPayload';
+
+// Define the list of possibly numerical properties to check and convert
+const propertiesToCheck = [
+  'versionNumber',
+  'siteUid',
+  'count',
+  'entity',
+  'amount',
+  'latitude',
+  'longitude',
+  'timestamp',
+  'bot',
+  'internalUser',
+  'isDirectAnswer'
+];
+
+export function convertTypesGTM(data: Record<string, unknown>): EventPayload {
```

Comment:
What exactly are we converting?

Reply:
User input from GTM always comes in as strings. We need to convert string represented numbers and booleans to the correct type before sending the request. This is what that takes care of.

Reply:
Then let's call this something like `convertStringtoValue` since GTM is not giving us some custom value that we are destructuring.